datasets/hind_encorp/hind_encorp.py

```diff
@@ -0,0 +1,173 @@
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        data_dir = dl_manager.download_and_extract(_URLs)
+        # data_paths = [os.path.join(pth, f)
+        # for pth, dirs, files in os.walk(data_dir) for f in files]
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir},
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        # for filepath in filepath:
+        with open(filepath, encoding="utf-8") as f:
+            id_ = 0
+            source = []
+            alignment_type = []
+            alignment_quality = []
+            English = []
+            Hindi = []
+            for line in f:
+                if line.startswith("-DOCSTART-") or line == "" or line == "\n" or line == "\t":
+                    if source:
+                        yield id_, {
+                            "id": str(id_),
+                            "source": source,
+                            "alignment_type": alignment_type,
+                            "alignment_quality": alignment_quality,
+                            "English": English,
+                            "Hindi": Hindi,
+                        }
+                        id_ += 1
+                        source = []
+                        alignment_type = []
+                        alignment_quality = []
+                        English = []
+                        Hindi = []
+                else:
+                    # conll2003 tokens are space separated
+                    splits = line.split("\t")
+                    source.append(splits[0])
+                    alignment_type.append(splits[1])
+                    alignment_quality.append(splits[2])
+                    English.append(splits[3])
+                    Hindi.append(splits[4])
+            # last exampleguid
+            yield id_, {
+                "id": str(id_),
+                "source": source,
+                "alignment_type": alignment_type,
+                "alignment_quality": alignment_quality,
+                "English": English,
+                "Hindi": Hindi,
+            }
```

Comment:
This looks wrong unfortunately
The data is composed of one pair of sentences per line, this is not the conll format.

Here is a suggestion of what it should look like in my opinion:

```python
            for id_, line in enumerate(f):
                splits = line.strip().split("\t")
                yield id_, {
                    "id": str(id_),
                    "source": splits[0],
                    "alignment_type": splits[1],
                    "alignment_quality": splits[2],
                    "English": splits[3],
                    "Hindi": splits[4],
                }
```

And another suggestion if you use the `Translation` feature type:

```python
            for id_, line in enumerate(f):
                splits = line.strip().split("\t")
                yield id_, {
                    "id": str(id_),
                    "source": splits[0],
                    "alignment_type": splits[1],
                    "alignment_quality": splits[2],
                    "translation": {
                        "en": splits[3],
                        "hi": splits[4]
                    },
                }
```